### DISCUSSION TITLE
IconPicker, partially configurable

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#7050] [Ideas]
Hi all, 

I have followed what is done with the `HtmlField` in `Trumbwyg` mode and I have created a version of the `TextField` in `IconPicker` mode that is partially configurable from the admin interface. 

I Attach a gif to allow you to see how it works.
![icon_picker](https://user-images.githubusercontent.com/2736009/92600331-b64c3b00-f2ab-11ea-841f-9a333c3869f0.gif)

If is something useful I could make a PR about it.
Thank you.

**Comment by Piedone** [comment]
Is it similar to what's in the Blog recipe for Taxonomies?

**Reply by PiemP** [reply]
I don't have seen the work done in the Blog recipe for Taxonomies. I have looked at the code for the `HtmlField` and copy it, to render this `IconPicker` customizable like we do with `HtmlField` when select the WYSWYG editor mode.

The `TextField` with the IconPicker editor type already exist in OrchardCore, but actually is static and doesn't provide the possibilities to cutosmize it (like limit the type of icons visible or use localization to customize the labels).

**Reply by Piedone** [reply]
When you set up with the Blog recipe you'll see that the Category term type has an icon:
![image](https://user-images.githubusercontent.com/1976647/93786238-5badca00-fc2f-11ea-840c-b7957c15b08c.png)

This is what I'm referring to.

**Reply by PiemP** [reply]
Sorry , I don't had understood. Yes, it's the same `IconPicker`. I have started from it to create"my version".

**Comment by deanmarcussen** [comment]
There is an issue here to extend the icon picker to support different icons https://github.com/OrchardCMS/OrchardCore/issues/3750.

It would be quite a nice feature to have at some point if you are interested in working on it @PiemP let us know and we could flesh out a design further.

The JSON editor for the HtmlField is used for settings because trumbowyg needs a JSON value for its javascript.

Normal or custom editor modes should probably use a typed class, rather than a JSON box for custom settings.

**Reply by deanmarcussen** [reply]
Actually I understand what you are asking now @PiemP .

You want to configure the javascript options for this script https://github.com/itsjavi/fontawesome-iconpicker

So editor settings like the trumbowyg would make sense.

But I'm not sure if we should include the option in OrchardCore, only because I see the source repository/script is now archived, and unmaintained by the owner.

We could wait until font awesome 6 is out to see if remains compatible before having a custom editor option for it.
